numba/cuda/cudadrv/driver.py

```diff
@@ -2229,41 +2686,115 @@ def complete(self):
         return cubin, size
 
 
-# -----------------------------------------------------------------------------
+class CudaPythonLinker(Linker):
+    """
+    Links for current device if no CC given
+    """
+    def __init__(self, max_registers=0, lineinfo=False, cc=None):
+        logsz = config.CUDA_LOG_SIZE
+        linkerinfo = bytearray(logsz)
+        linkererrors = bytearray(logsz)
 
+        jit_option = cuda_driver.CUjit_option
 
-def _device_pointer_attr(devmem, attr, odata):
-    """Query attribute on the device pointer
-    """
-    error = driver.cuPointerGetAttribute(byref(odata), attr,
-                                         device_ctypes_pointer(devmem))
-    driver.check_error(error, "Failed to query pointer attribute")
+        options = {
+            jit_option.CU_JIT_INFO_LOG_BUFFER: linkerinfo,
+            jit_option.CU_JIT_INFO_LOG_BUFFER_SIZE_BYTES: logsz,
+            jit_option.CU_JIT_ERROR_LOG_BUFFER: linkererrors,
+            jit_option.CU_JIT_ERROR_LOG_BUFFER_SIZE_BYTES: logsz,
+            jit_option.CU_JIT_LOG_VERBOSE: 1,
+        }
+        if max_registers:
+            options[jit_option.CU_JIT_MAX_REGISTERS] = max_registers
+        if lineinfo:
+            options[jit_option.CU_JIT_GENERATE_LINE_INFO] = 1
 
+        if cc is None:
+            # No option value is needed, but we need something as a placeholder
+            options[jit_option.CU_JIT_TARGET_FROM_CUCONTEXT] = 1
+        else:
+            cc_val = cc[0] * 10 + cc[1]
+            cc_enum = getattr(cuda_driver.CUjit_target,
+                              f'CU_TARGET_COMPUTE_{cc_val}')
+            options[jit_option.CU_JIT_TARGET] = cc_enum
 
-def device_pointer_type(devmem):
-    """Query the device pointer type: host, device, array, unified?
-    """
-    ptrtype = c_int(0)
-    _device_pointer_attr(devmem, enums.CU_POINTER_ATTRIBUTE_MEMORY_TYPE,
-                         ptrtype)
-    map = {
-        enums.CU_MEMORYTYPE_HOST: 'host',
-        enums.CU_MEMORYTYPE_DEVICE: 'device',
-        enums.CU_MEMORYTYPE_ARRAY: 'array',
-        enums.CU_MEMORYTYPE_UNIFIED: 'unified',
-    }
-    return map[ptrtype.value]
+        raw_keys = list(options.keys())
+        raw_values = list(options.values())
+
```

Comment:
I can't see why that would be, and the `del` goes back to the first commit opensourcing NumbaPro. I'm going to try deleting it.

Reply:
I've now removed it, I don't think it helped with anything.

Reply:
I'm not sure what the intended effect was? I guess it's lost in time. I do note however that there's a `self._keep_alive` being assigned to locally and wonder if it's related to that? Deleting `options` would remove reference to the keys/values that are stored in the `raw_` vars that end up in `_keep_alive`. It's not clear though why this might matter, once the link is complete and these containers are deleted, some out of scope prior reference is unlikely to have any impact?

Reply:
I think it could have been related to that, but without the source to NumbaPro its hard to work out the history. The `_keep_alive` bits are still necessary at present. I don't think there's anything to change here.

Reply:
ok, let's leave it until something proves/demonstrates it as a problem otherwise. Thanks for taking another look.